### PR TITLE
Add getAppDataDoc method

### LINKED ISSUE #960: [Affiliate] Implement api to get metadata doc
# Summary

Implement api to get metadata doc similar to 
https://github.com/gnosis/cowswap/issues/956

It should send a GET request to (still to be finalized) `/appData/<address>`

The response will be the metadata doc and hash (same data send as input on #956) when it exists (status code `200`) or empty response when it doesn't (status code `404`).

The function should return null instead of error on `404`.

# Requirements

- New backend api function to get metadata doc, if any
- Implement mocks if backend is not ready in time

# Out of scope

- 

# Additional context

- Wireframes: https://www.figma.com/file/P0AIwKkIUpEZtLukkCUM94/CowSwap_-Affiliate?node-id=48%3A224
- Design doc: https://docs.google.com/document/d/1PsdWn9WPFf2BNeoJNtClyra8jNvWk_bJ-g33TbpUf8s/edit#

### PR DESCRIPTION
**Pull request by ramirotw**
# Summary

Closes #960 

Adds a `getAppDataDoc` operator function to fetch the affiliate metadata given a referral address. I also added a mocked version and renamed the `MOCK` env variable to `REACT_APP_MOCK`.

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]
LGTM!

We'll soon have this ready and can close the open questions :)

File: src/custom/utils/operator/errors/MetadataError.ts [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
   description: string
 }
 
 // TODO: Validate/Update error code when the API is updated
-export enum ApiErrorCodes {
+export enum MetadataApiErrorCodes {
+  AddressNotFound = 'AddressNotFound',
+  UNHANDLED_GET_ERROR = 'UNHANDLED_GET_ERROR',
   UNHANDLED_UPLOAD_ERROR = 'UNHANDLED_UPLOAD_ERROR',
 }
```
Comment (stackseeder) [inline src/custom/utils/operator/errors/MetadataError.ts:13]: Is there any reason why we are using different cases for enum variables?
How about `ADDRESS_NOT_FOUND`, unless it makes anything wrong?

**Review by matextrem — APPROVED** [review]

File: src/custom/utils/operator/errors/MetadataError.ts [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
 
-export enum ApiErrorCodeDetails {
+export enum MetadataApiErrorCodeDetails {
+  AddressNotFound = "The referral address doesn't exist",
+  UNHANDLED_GET_ERROR = 'Metadata AppDoc fetch failed. This may be due to a server or network connectivity issue. Please try again later.',
   UNHANDLED_UPLOAD_ERROR = 'The metadata object could not be saved',
```
Comment (stackseeder) [inline src/custom/utils/operator/errors/MetadataError.ts:18]: The same question as above.

File: src/custom/utils/operator/index.ts [hunk 3]
```diff
@@ -2,14 +2,16 @@ export type { UnsupportedToken, OrderID, OrderMetaData } from './operatorApi'
 import * as realApi from './operatorApi'
 import * as mockApi from './operatorMock'
 
-const useMock = process.env.MOCK_OPERATOR === 'true'
+const useMock = process.env.REACT_APP_MOCK === 'true'
+console.log({ useMock })
```
[the remark this answers is not in the document]
Comment (ramirotw) [reply src/custom/utils/operator/index.ts]: yes, I already removed it

File: src/custom/utils/operator/errors/MetadataError.ts (hunk 1 above)
Comment (ramirotw) [reply src/custom/utils/operator/errors/MetadataError.ts:13]: I followed the existing conventions. Take a look at `OperatorError` and `QuoteError` files

**Pull request merged**